API.md

```diff
@@ -0,0 +1,518 @@
+
+```json
+{
+  "id": 1,
+  "meta": {
+    "lastUpdated": "2020-05-11T02:01:16+00:00"
+  },
+  "status": "final",
+  "subject": {
+    "reference": "Patient/1"
+  },
+  "effectiveDateTime": "2020-05-11T00:00:00+00:00",
+  "valueString": "positive",
+  "resourceType": "Observation"
+}
+```
+
+<a name="read-get-que"/>
+
+### GET `[base]/QuestionnaireResponse/[:id]`
+
+Get a monitoree daily report via an id, e.g.:
+
+```json
+{
+  "id": 3,
+  "meta": {
+    "lastUpdated": "2020-05-11T01:06:37+00:00"
+  },
+  "status": "completed",
+  "subject": {
+    "reference": "Patient/1"
+  },
+  "item": [
+    {
+      "text": "cough",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    },
+    {
+      "text": "difficulty-breathing",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    },
+    {
+      "text": "fever",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    },
+    {
+      "text": "used-a-fever-reducer",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    },
+    {
+      "text": "chills",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    },
+    {
+      "text": "repeated-shaking-with-chills",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    },
+    {
+      "text": "muscle-pain",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    },
+    {
+      "text": "headache",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    },
+    {
+      "text": "sore-throat",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    },
+    {
+      "text": "new-loss-of-taste-or-smell",
+      "answer": [
+        {
+          "valueBoolean": false
+        }
+      ]
+    }
+  ],
+  "resourceType": "QuestionnaireResponse"
+}
+```
+
+<a name="read-get-all"/>
+
+### GET `[base]/Patient/[:id]/$everything`
+
+Use this route to retrieve a FHIR Bundle containing the monitoree, all their lab results, and all their daily reports.
+
+```json
+{
+  "id": "9e02e507-b1c0-4fda-ac40-73dab0bd7dc9",
+  "meta": {
+    "lastUpdated": "2020-05-11T22:12:22-04:00"
+  },
+  "type": "searchset",
+  "total": 8,
+  "entry": [
+  ],
+  "resourceType": "Bundle"
+}
+```
+
+<a name="create"/>
+
+## Creating
+
+The API supports creating new monitorees.
+
+<a name="create-post-pat"/>
+
+### POST `[base]/Patient`
+
+#### Request Body
+
+```json
+{
+  "resourceType": "Patient",
+  ...
+}
+```
+
+#### Response
+
+On success, the server will return the newly created resource with an id. This is can be used to retrieve or update the record moving forward.
+
+```json
+{
+  "resourceType": "Patient",
+  "id": 1,
+  ...
+}
+```
+
+<a name="update"/>
+
+## Updating
+
+The API supports updating existing monitorees.
+
+<a name="update-put-pat"/>
+
+### PUT `[base]/Patient/[:id]`
```

Comment:
I ran some updates on `Patient` where I was able to remove certain data, such as OMB Race. What fields are required or must-support for this API? Does it matter if Race or Birthsex or any field is empty?

Reply:
I was able to successfully update a Patient with the following body:
```json
{
  "id": 1017,
  "resourceType": "Patient"
}
```

Should this be allowed? There is no longer a name, phone number, or address.

Reply:
We're not requiring any fields at the moment in the flat-file import, since jurisdictions asked for this. Although at some point I do think we need to enforce some level of required fields. That being said, the application will handle blank fields like this fine. I do not think you should be able to blank out fields though, so I stopped that from happening. Willing to discuss further.